src/ResponseTagger.php

```diff
@@ -32,37 +35,41 @@ class ResponseTagger
     private $options;
 
     /**
-     * @var TagCapable
+     * @var TagHeaderFormatter
      */
-    private $proxyClient;
+    private $headerFormatter;
 
     /**
      * @var array
      */
     private $tags = [];
 
     /**
-     * Create the response tagger with a tag capable proxy client and options.
+     * Create the response tagger with a tag header formatter and options.
      *
      * Supported options are:
      *
+     * - header_formatter (TagHeaderFormatter) Default: CommaSeparatedTagHeaderFormatter with default header name
      * - strict (bool) Default: false. If set to true, throws exception when adding empty tags
      *
-     * @param TagCapable $proxyClient
-     * @param array      $options
+     * @param array $options
      */
-    public function __construct(TagCapable $proxyClient, array $options = [])
+    public function __construct(array $options = [])
     {
-        $this->proxyClient = $proxyClient;
-
         $resolver = new OptionsResolver();
         $resolver->setDefaults([
+            // callback to avoid instantiating the formatter when its not needed
+            'header_formatter' => function (Options $options) {
+                return new CommaSeparatedTagHeaderFormatter();
```

Comment:
@stof is this a good idea? i wanted to avoid the overhead of creating an object when none is needed (because the formatter is provided by the user). the documentation on options resolver mentions the callback with an Options argument, but does not mention this use case explicitly.